idaes/power_generation/control/pid_controller.py

```diff
@@ -0,0 +1,208 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2019, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+PID controller block, Revised to use pyomo.DAE
```

Comment:
A general question ,but how does this differ from the PID controller in the core library?

Reply:
This one has a variable for the integral error term.  The core library now uses the velocity form to avoid the need for it, but I'm not sure why one may be better than the other and when to prefer one over the other.  This also uses different functions to limit the controller output between bounds.  Also this supports P and PD.

Eventually when I have time, I'll merge those things in as options to the core controller.  In the interest of time, we are going to hold off on the merger to a single PID class.